database/src/main/java/com/firebase/ui/database/FirebaseListAdapter.java

```diff
@@ -50,11 +50,10 @@
  * @param <T> The class type to use as a model for the data contained in the children of the given Firebase location
  */
 public abstract class FirebaseListAdapter<T> extends BaseAdapter {
-
-    private final Class<T> mModelClass;
-    protected int mLayout;
     protected Activity mActivity;
-    FirebaseArray mSnapshots;
+    protected final Class<T> mModelClass;
+    protected int mLayout;
+    protected FirebaseArray mSnapshots;
```

Comment:
The `mSnapshots` field should not be modified by subclasses, hence it should not be accessible. All the pertinent info of snapshots is already available via `getItem()`, `getRef()` and `getRef()`.